File: Java/1630_Arithmetic_Subarrays.java
class Solution {
    public List<Boolean> checkArithmeticSubarrays(int[] nums, int[] l, int[] r) {
        List<Boolean> result = new ArrayList<>();
        int m = l.length;
        
        for(int i = 0; i < m; i++){
            int[] temp = new int[r[i] - l[i]+1];
            for(int j = 0; j < temp.length; j++){
                temp[j] = nums[l[i]+j];
            }
            Arrays.sort(temp);
            int d = temp[1] - temp[0];
            boolean f = true;
            for(int k = 1; k < temp.length; k++){
                if(temp[k] - temp[k-1] != d){
                    f = false;
                    break;
                }
            }
            result.add(f);
        }
        
        return result;
    }
}

File: CS/1630_Arithmetic_Subarrays.cs
public class Solution {
    public IList<bool> CheckArithmeticSubarrays(int[] nums, int[] l, int[] r) {
        IList<bool> result = new List<bool>();
        for(int i = 0; i < l.Length; i++){
            bool f = true;
            int[] arr = new int[r[i] - l[i] + 1];
            for(int j = 0; j < arr.Length; j++){
                arr[j] = nums[l[i]+j];
            }
            Array.Sort(arr);
            for(int j = 1; j < arr.Length-1; j++){
                if(arr[j] - arr[j-1] != arr[j+1] - arr[j]){
                    f = false;
                    break;
                }
            }
            result.Add(f);
        }
        return result;
    }
}
